junit-platform-engine/src/main/java/org/junit/platform/engine/support/hierarchical/ExclusiveResource.java

```diff
@@ -52,18 +52,34 @@ public class ExclusiveResource {
 
 	private final String key;
 	private final LockMode lockMode;
+	private final LockScope lockScope;
 	private int hash;
 
 	/**
-	 * Create a new {@code ExclusiveResource}.
+	 * Create a new {@code ExclusiveResource} with a default lock scope {@link LockScope#SELF}
 	 *
 	 * @param key the identifier of the resource; never {@code null} or blank
 	 * @param lockMode the lock mode to use to synchronize access to the
 	 * resource; never {@code null}
+	 *
 	 */
 	public ExclusiveResource(String key, LockMode lockMode) {
+		this(key, lockMode, LockScope.SELF);
+	}
+
+	/**
+	 * Create a new {@code ExclusiveResource}.
+	 *
+	 * @param key the identifier of the resource; never {@code null} or blank
+	 * @param lockMode the lock mode to use to synchronize access to the
+	 * resource; never {@code null}
+	 * @param lockScope the lock scope to use to synchronize access to the
+	 * resource; never {@code null}
+	 */
+	public ExclusiveResource(String key, LockMode lockMode, LockScope lockScope) {
```

Comment:
I don't think this should be a Platform concept. Instead, Jupiter should return `ExclusiveResource` instances from the children `TestDescriptors` if their parent declares a `@ResourceLock` with `target` of `CHILDREN`.

Reply:
sounds good, no need to bother Platform with those details :) I drafted how it may look in https://github.com/junit-team/junit5/pull/3220/files#diff-63a01db8dd78b02e05c3888eb786c656af0cd3dc8263c816530003f2d7069daeR188